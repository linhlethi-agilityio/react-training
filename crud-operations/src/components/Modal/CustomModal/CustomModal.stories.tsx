import type { Meta, StoryObj } from '@storybook/react';
import { Box } from '@chakra-ui/react';

// Components
import CustomModal from '.';

const meta: Meta<typeof CustomModal> = {
  component: CustomModal,
};

export default meta;
type Story = StoryObj<typeof CustomModal>;

export const Default: Story = {
  args: {
    isOpen: true,
    onClose: () => null,
    title: 'Student Detail',
    children: (
      <Box>
        Lorem ipsum dolor, sit amet consectetur adipisicing elit. Deserunt quasi veritatis id aperiam delectus iste
        aliquid fugit sint ea ex? Aliquid totam blanditiis quae quia repellat eos unde consequuntur laborum!
      </Box>
    ),
  },
};
