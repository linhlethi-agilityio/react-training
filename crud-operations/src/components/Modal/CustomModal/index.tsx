import { ReactNode, memo } from 'react';
import { Modal, ModalBody, ModalCloseButton, ModalContent, ModalHeader, ModalOverlay } from '@chakra-ui/react';

interface CustomModalProps {
  isOpen: boolean;
  onClose: () => void;
  title: string;
  children: ReactNode;
}

const CustomModal = ({ isOpen, children, title, onClose }: CustomModalProps) => (
  <Modal isOpen={isOpen} onClose={onClose} isCentered>
    <ModalOverlay />
    <ModalContent opacity={0.6}>
      <ModalHeader textAlign="center">{title}</ModalHeader>
      <ModalCloseButton />
      <ModalBody>{children}</ModalBody>
    </ModalContent>
  </Modal>
);

export default memo(CustomModal);
