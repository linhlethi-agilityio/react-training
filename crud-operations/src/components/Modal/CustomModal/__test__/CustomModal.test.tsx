import { render } from '@test-utils';
import '@testing-library/jest-dom';
import { Box } from '@chakra-ui/react';

// Components
import CustomModal from '..';

const mockProps = {
  isOpen: true,
  onClose: jest.fn(),
  title: 'Student Detail',
  children: (
    <Box>
      Lorem ipsum dolor, sit amet consectetur adipisicing elit. Deserunt quasi veritatis id aperiam delectus iste
      aliquid fugit sint ea ex? Aliquid totam blanditiis quae quia repellat eos unde consequuntur laborum!
    </Box>
  ),
};

describe('CustomModal component', () => {
  test('should match snapshot for CustomModal', () => {
    const { container } = render(<CustomModal {...mockProps} />);

    expect(container).toMatchSnapshot();
  });
});
