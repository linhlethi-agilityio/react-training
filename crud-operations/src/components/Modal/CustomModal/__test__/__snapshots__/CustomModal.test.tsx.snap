// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomModal component should match snapshot for CustomModal 1`] = `
<div
  aria-hidden="true"
  data-aria-hidden="true"
/>
`;
